scm-webapp/src/test/java/sonia/scm/lifecycle/AdminAccountStartupActionTest.java

```diff
@@ -0,0 +1,132 @@
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package sonia.scm.lifecycle;
+
+import com.google.common.collect.Lists;
+import org.apache.shiro.authc.credential.PasswordService;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.junit.jupiter.MockitoSettings;
+import org.mockito.quality.Strictness;
+import sonia.scm.SCMContext;
+import sonia.scm.security.PermissionAssigner;
+import sonia.scm.security.PermissionDescriptor;
+import sonia.scm.user.User;
+import sonia.scm.user.UserManager;
+import sonia.scm.user.UserTestData;
+
+import java.util.Collection;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class AdminAccountStartupActionTest {
+
+  @Mock
+  private PasswordService passwordService;
+  @Mock
+  private UserManager userManager;
+  @Mock
+  private PermissionAssigner permissionAssigner;
+
+  @InjectMocks
+  private AdminAccountStartupAction startupAction;
+
+  @Test
+  void shouldCreateAdminAccountIfNoUserExistsAndAssignPermissions() {
+    when(passwordService.encryptPassword("scmadmin")).thenReturn("secret");
+
+    startupAction.run();
+
+    verifyAdminCreated();
+    verifyAdminPermissionsAssigned();
+  }
+
+  @Test
+  void shouldCreateAdminAccountIfOnlyAnonymousUserExistsAndAssignPermissions() {
+    when(userManager.getAll()).thenReturn(Lists.newArrayList(SCMContext.ANONYMOUS));
+    when(userManager.contains(SCMContext.USER_ANONYMOUS)).thenReturn(true);
+    when(passwordService.encryptPassword("scmadmin")).thenReturn("secret");
+
+    startupAction.run();
+
+    verifyAdminCreated();
+    verifyAdminPermissionsAssigned();
+  }
+
+  @Test
+  @MockitoSettings(strictness = Strictness.LENIENT)
+  void shouldSkipAdminAccountCreationIfPropertyIsSet() {
+    System.setProperty("sonia.scm.skipAdminCreation", "true");
+
+    try {
+      startupAction.run();
+
+      verify(userManager, never()).create(any());
+      verify(permissionAssigner, never()).setPermissionsForUser(anyString(), any(Collection.class));
```

Comment:
Specifying the class is not necessary here (and avoids a warning, too)
```suggestion
      verify(permissionAssigner, never()).setPermissionsForUser(anyString(), any());
```